archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/internal/ArchUnitTestEngine.java

```diff
@@ -67,6 +72,22 @@
 public final class ArchUnitTestEngine extends HierarchicalTestEngine<ArchUnitEngineExecutionContext> {
     static final String UNIQUE_ID = "archunit";
 
+    private static final Collection<String> ABORTING_THROWABLE_NAMES = Arrays.asList(
+            "org.junit.internal.AssumptionViolatedException",
```

Comment:
Why would we support JUnit 4 exceptions within the JUnit 5 support? :thinking: Besides that `.internal.` anyway sounds in a way that we should not handle it :man_shrugging:

Reply:
Those are all the flavors of assumption violated exceptions that can be thrown to indicate assumption violation (i.e. exceptions that should result in skip results, not failures). 

As to whether we should be handling the exception or not, please take a look at JUnit's [`OpenTest4JAndJUnit4AwareThrowableCollector`](https://github.com/junit-team/junit5/blob/main/junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/support/OpenTest4JAndJUnit4AwareThrowableCollector.java). 

> Why would we support JUnit 4 exceptions within the JUnit 5 support?

Remember that you can run JUnit 4 ArchUnit tests using the `ArchUnitEngine` (NOT Vintage!)

Reply:
> JUnit 4 support has a different annotation @AnalyzeClasses than JUnit 5 support has.

Different *how*? They have the same name and package. 

This is exactly the configuration issue I was trying to tell you about. If you run your tests with both archunit and vintage engines, the tests are being run twice. They are eligible for discovery by both engines. Try debugging the tooling tests with both engines on, you will see both engines executing the test cases. 

> I don't think users should use JUnit 4 things within the scope of the ArchUnitTestEngine.

The question is do users know that? If they use the method flavor of archunit tests, they are allowed to use `assertThat`. Why wouldn't they be allowed to use `assumeThat`?

Reply:
> it is not supported to use archunit-junit4 and archunit-junit5 together on the classpath

Well, I mean... that changes things. A lot :p 

I was talking about the scenario where you have a project with JUnit 4 arch tests, with both archunit and junit-vintage engines enabled. In that case, each test gets picked up by junit-vintage (because of @RunWith(ArchunitRunner.class)) *and* by archunit (because of @AnalyzeClasses). You can't have both ArchUnitRunner *and* archunit engine on the classpath, though, while not simultaneously depending on both archunit-junit dependencies. So that makes sense to me now. 

Where exactly do the docs say that mixing archunit-junit4 and archunit-junit5 is not supported, btw? I read the user guide on junit support, but I couldn't find it anywhere...

In that case, I now realize I wrongly assumed support for test filtering was working with junit-vintage when actually it just happened to work because the archunit engine was the one that was running the tests :(

...which means there is no point in any support for junit 4 stuff in this or any of the other PRs. I will try to update the PRs accordingly next week. Thanks for your patience:)